gboard_autopsy.py

```diff
@@ -244,6 +256,27 @@ def report_analysis(self, input_dir, data_source, blackboard, file_manager, anal
                     (self.dictionary_locale_attr_type, entry['locale'])
                 ])
 
+        for trainingcache in analysis_output['trainingcache']:
+            file = self.get_common_sufix_file(data_source, file_manager, input_dir, trainingcache['path'])
+            # process raw histories
+            for entry in trainingcache['rawHistory']:
+                self.publish_raw_history_artifact(blackboard, file, entry, self.tc_raw_history_art_type)
+            for entry in trainingcache['rawAssembledHistory']:
+                self.publish_raw_history_artifact(blackboard, file, entry, self.tc_raw_ahistory_art_type)
+            # process relevant histories
+            for entry in trainingcache['relevantHistory']:
+                self.publish_analysis_artifact(blackboard, file, self.tc_relevant_history_art_type, [
+                    (BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DATETIME, entry['timestamp']),
+                    (BlackboardAttribute.ATTRIBUTE_TYPE.TSK_TEXT, entry['sequence']),
+                ])
```

Comment:
Maybe we can optimize this to `O(N)`. Can't we create a tuple of arrays with all those Dict keys and, in combination with `Option`, iterate everything until all outputs are `None`?